.github/workflows/tag.yml

```diff
@@ -0,0 +1,37 @@
+name: Create Tag
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "Tag name, defaults to minor increment."
+        required: false
+      sha:
+        description: "Commit SHA to tag. Defaults to latest."
+        required: false
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set the commit SHA
+        id: sha
+        run: echo "::set-output name=commitSha::$(if [ -z "$SHA" ]; then echo $GITHUB_SHA; else echo $SHA; fi)"
+        env:
+          SHA: ${{ github.event.inputs.sha }}
+      - name: Set the tag
+        id: tag
+        run: |
+          last_tag="$(git describe --tags --abbrev=0)"
+          new_tag="$(echo "$last_tag" | perl -pe 's/^(v\d+\.)(\d+)(\.\d+)$/$1.($2+1).($3)/e')"
+          echo "::set-output name=newTag::$(if [ -z "$TAG" ]; then echo $new_tag; else echo $TAG; fi)"
+        env:
+          TAG: ${{ github.event.inputs.tag }}
+      - name: Create tag
+        uses: actions/github-script@v5
+        with:
+          script: |
+            github.rest.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: 'refs/tags/${{ steps.tag.outputs.newTag }}',
```

Comment:
If we're going to support passing in an arbitrary tag, will this fail if we're re-using a tag that already exists?

Reply:
Maybe? :)

Reply:
Cool yeah if not let's add a failure here to catch that case and avoid accidentally stomping over an existing release.